### ISSUE TITLE
Weird behaviour with chaining?

### ISSUE THREAD

**Issue by mathieutu** [prettier/plugin-php#607]
I don't know if it's normal or not, but I find that weird...
What are your thoughts?

```diff
-        return $this->resources($hotel->locations()->withTrashed()->get());
+        return $this->resources(
+            $hotel
+                ->locations()
+                ->withTrashed()
+                ->get()
+        );

```

**Comment by evilebottnawi** [comment]
@mathieutu look expected, same for prettier https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAnOMCuqoAIAkMAFgJYDOAdOmRNmHGQBT5ETwA2F7EYAhjCWhMAlBQDuJYgBVUvMkTgATRqIDmGFcIDcIADQgIABwFDkoXqlQQxABQsIyyELwBuEEor0gARrLABrDABlQ14wEihVZBhUTDh9CLI4VBgbWVUAW15kADNediT9ACsyAA8AIT9AmCDeDLgAGQi4XPzCkFDUJNQnb15vAE9uKC9DVAiYAHUPYmQADgAGfTGIJMnZQycxhmSXFv10AEdMEnQ03kzspDyC+JAkjJJo2LuyCNV2OABFTDYW67adxg-WmilmSAATPoYrwSOx3gBhCAZLJOKDQfYgTBJKT9RwA24AX0JQA

**Comment by mathieutu** [comment]
Ok, I don't find this obvious...

Same here, why does it break just after `orders()` ?
```diff
-        $orders_without_brian = $brian->hotel->orders()->ordered()->with('smith')->get();
+        $orders_without_brian = $brian->hotel->orders()
+            ->ordered()
+            ->with('smith')
+            ->get();
```

**Comment by evilebottnawi** [comment]
Hm, looks ugly, prettier https://prettier.io/playground/#N4Igxg9gdgLgprEAuEACdASCAnAJnbAZwH0B3ASxgAsIBXGYgI23IEMpUBeVDZtqAHQ14AGwE58RABQBKADod0S5enF4CcXLIUrdqARWpSA5IQC2lKsfmK9SgQHM4MWQG4QAGhAQADjHLQhMigrNjYEKQACqEIQSisAG4Q5LieIMysYADWzgDKPpnkUA7IMNi0cF5FhAQwkdisDmasyABmrCI1XgBWhAAeAEIN2XmsZnAAMkVwbR1dIAVEBMjprIwAniLQaT4ssADqKdTIABwADF67EDX7DT4ru3A12AkzXthwAI605B-1jc1Zp1KiAahZSuUQYQig4RHAAIq0CDwIHzGBrQ64Y5IABMXjKrHIIhhAGEIGZASgoNA3iBaDUACprOLtYEAXzZQA

**Comment by mathieutu** [comment]
Yeah I've tried it in prettier before, but the prettier way is not really better.

Another one to make my point that the previous one is better than the prettier way: 

```diff
             'order_week_quantity' => (clone $q)
-                ->whereBetween('delivered_at', [now()->subWeek()->startOfDay(), now()])
+                ->whereBetween('delivered_at', [
+                    now()
+                        ->subWeek()
+                        ->startOfDay(),
+                    now(),
+                ])
```

**Comment by mathieutu** [comment]
I've plenty of examples 😅.
For me here the function should be in one block, so on its own line, like before prettier pass.

```diff
-            'average_order_delivering_time' => $q->get(['created_at', 'delivered_at'])
-                ->average(function (Order $order) {
-                    return $order->delivered_at->diffInSeconds($order->created_at);
-                }),
+            'average_order_delivering_time' => $q->get(['created_at', 'delivered_at'])->average(function (Order $order) {
+                return $order->delivered_at->diffInSeconds($order->created_at);
+            }),
```

**Comment by evilebottnawi** [comment]
@mathieutu we use `prettier` logic for `php`, if you find something weird try first on playground https://prettier.io/playground/ with `js`, maybe some problems need fix in prettier first.

**Comment by mathieutu** [comment]
Yeah, but as you said php logic is not the same than js one.
It's not because it's ok in js, that it should be the same in php.


Other example for the record (just in case we have to debate later):
```diff
-        if (!Brian::whereHotelId($hotelId)->available()->exists()) {
+        if (
+            !Brian::whereHotelId($hotelId)
+                ->available()
+                ->exists()
+        ) {
```

**Comment by evilebottnawi** [comment]
@mathieutu and what you expected?

**Comment by evilebottnawi** [comment]
@mathieutu please provide example where you disagree with output and you configuration line length

**Comment by czosel** [comment]
> Yeah, but as you said php logic is not the same than js one.
It's not because it's ok in js, that it should be the same in php.

While that is correct, we try to stay close to the JS implementation for anything that maps well to PHP, because prettier for JS is already quite mature and usually at lot of thought went into their formatting decisions. I think the decision to format chained method calls like we do it now maps well from JS to PHP. The only actionable thing I see in this issue is the example 
```php
$orders_without_brian = $brian->hotel->orders()
            ->ordered()
            ->with('smith')
            ->get();
```
which should be formatted the same way as prettier for JS.

**Comment by evilebottnawi** [comment]
@mathieutu if you still find something weird please create new issue, better one example output - one issue, it is more comfortable, thanks for using `prettier`
